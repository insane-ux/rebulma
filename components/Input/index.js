// @flow

import React from 'react'
import { themr } from 'react-css-themr'
import Icon from 'components/Icon'
import cn from 'classnames'

export const defaultStyles = {
  root: '',
  input: '',
  leftIcon: '',
  rightIcon: '',
}

const Input = ({
  theme,
  className,
  leftIcon,
  rightIcon,
  ...props
}: {
  theme: { [key: string]: string },
  className?: string,
  leftIcon?: string,
  rightIcon?: string,
}) => {
  const rootClassName = cn('control', className, theme.root, {
    'has-icons-left': !!leftIcon,
    'has-icons-right': !!rightIcon,
  })
  return (
    <p className={rootClassName}>
      <input
        className={cn('input', theme.input)}
        type="text"
        {...props}
      />
      {leftIcon && (
        <Icon
          name={leftIcon}
          className={cn('is-left', theme.leftIcon)}
        />
      )}
      {rightIcon && (
        <Icon
          name={rightIcon}
          className={cn('is-right', theme.rightIcon)}
        />
      )}
    </p>
  )
}

Input.defaultProps = {
  className: undefined,
  leftIcon: undefined,
  rightIcon: undefined,
}

export default themr('Input', defaultStyles)(Input)
