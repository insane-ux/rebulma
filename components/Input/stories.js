// @flow

import React from 'react'
import { storiesOf } from '@kadira/storybook'
import Input from './'

storiesOf('Input', module)
  .add('default', () => (
    <Input />
  ))
  .add('small', () => (
    <Input theme={{ input: 'is-small' }} />
  ))
  .add('with left icon', () => (
    <Input leftIcon="fa fa-pencil-square-o" theme={{ input: 'is-large' }} />
  ))
  .add('with right icon', () => (
    <Input rightIcon="fa fa-pencil-square-o" theme={{ input: 'is-small', leftIcon: 'is-small' }} />
  ))
  .add('with both icon', () => (
    <Input
      leftIcon="fa fa-pencil-square-o"
      rightIcon="fa fa-check"
      theme={{ leftIcon: 'is-small' }}
    />
  ))
