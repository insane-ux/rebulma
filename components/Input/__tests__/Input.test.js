/* global it, expect, describe */

// @flow

import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import Input from '../index'

describe('Input', () => {
  it('Input with className', () => {
    const input = mount(
      <Input theme={{ input: 'is-small' }} />,
    )
    expect(input.find('.is-small').prop('className')).toEqual('input is-small')
  })

  it('Input with className', () => {
    const component = renderer.create(<Input theme={{ input: 'is-small' }} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
