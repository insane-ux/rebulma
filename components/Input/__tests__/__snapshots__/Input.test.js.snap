// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Input with className 1`] = `
<p
  className="control"
>
  <input
    className="input is-small"
    placeholder="Text input"
    type="text"
  />
</p>
`;
